perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonScopedAffiliations.java

```diff
@@ -107,12 +108,46 @@ public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDe
 	}
 
 	@Override
-	public List<Pattern> getPatternsForMatch() {
-		List<Pattern> patterns = super.getPatternsForMatch();
-		patterns.add(userAllAttrsRemovedPattern);
-		patterns.add(userEPSAMARemovePattern);
-		patterns.add(userEPSAMASetPattern);
+	public List<AuditerMessage> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, AuditerMessage message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<AuditerMessage> resolvingMessages = super.resolveVirtualAttributeValueChange(perunSession, message);
+
+		if (message.getEvent() instanceof AllAttributesRemovedForUser) {
+			User user = ((AllAttributesRemovedForUser) message.getEvent()).getUser();
```

Comment:
I would like to see this parsed the same way as in urn_perun_member_resource_attribute_def_virt_isBanned.java module to private methods. The same for all other modules.